{{cookiecutter.project_slug}}/fabfile.py

```diff
@@ -213,180 +343,199 @@ def bootstrap():
+        try:
+            c.conn.run("test -d .", hide=True)
+        except UnexpectedExit:
+            raise Exit(
+                "Provisioning not finished, directory {} doesn't exist!".format(
+                    c.config["root"]
+                )
+            )
+        # Now make sure there's git, and a git repository
+        try:
+            c.conn.git("--version", hide=True)
+        except UnexpectedExit:
+            raise Exit("Provisioning not finished, git not available!")
+
+        try:
+            c.conn.git("rev-parse --git-dir", hide=True)
+        except UnexpectedExit:
+            c.conn.git("init")
+            c.conn.git("checkout --orphan master")
+            c.conn.put(StringIO("Bootstrap"), os.path.join(c.conn.wwwroot, "bootstrap"))
+            c.conn.git("add bootstrap")
+            c.conn.git('commit -m "First non-empty commit"')
+
+    git_remote_url = "ssh://{user}@{host}:{port}/{directory}".format(
+        user=c.conn.user, host=c.conn.host, port=c.conn.port, directory=c.conn.wwwroot
     )
 
+    # Now push our code to the remote, always as FABHEAD branch
+    porcelain.push(".", git_remote_url, "{}:FABHEAD".format(git_ref))
 
-@task
-def deploy():
-    require("root", "project_name")
-
-    execute(git_push, commit="@")
-    dump_db(get_backups_root())
-
-    execute(install_requirements)
-    execute(compile_assets)
-    execute(collect_static)
-    execute(migrate_database)
+    with c.conn.cd(c.conn.wwwroot):
+        c.conn.git("checkout -f master", hide=True)
+        c.conn.git("reset --hard FABHEAD")
+        c.conn.git("branch -d FABHEAD", hide=True)
+        c.conn.git("submodule update --init", hide=True)
 
-    execute(restart_process)
-    execute(clean_old_database_backups, nb_backups_to_keep=10)
 
-
-def dump_db(destination):
+@task
+@remote
+def install_requirements(c):
     """
-    Dump the database to the given directory and return the path to the file created. This creates a gzipped SQL file.
+    Install project requirements in venv
     """
-    with cd(get_project_root()), quiet():
-        db_credentials = run("cat envdir/DATABASE_URL")
-    db_credentials_dict = dj_database_url.parse(db_credentials)
-
-    if not is_supported_db_engine(db_credentials_dict["ENGINE"]):
-        raise NotImplementedError(
-            "The dump_db task doesn't support the remote database engine"
-        )
-
-    outfile = os.path.join(
-        destination, datetime.now().strftime("%Y-%m-%d_%H%M%S.sql.gz")
-    )
-
-    with shell_env(PGPASSWORD=db_credentials_dict["PASSWORD"].replace("$", "\$")):
-        run(
-            "pg_dump -O -x -h {host} -U {user} {db}|gzip > {outfile}".format(
-                host=db_credentials_dict["HOST"],
-                user=db_credentials_dict["USER"],
-                db=db_credentials_dict["NAME"],
-                outfile=outfile,
-            )
-        )
+    try:
+        # Test that pip works
+        c.conn.pip("freeze", hide=True)
```

Comment:
Couldn't we instead test if the virtualenv directory exists? It would be faster.